src/EnergyPlus/UnitarySystem.cc

```diff
@@ -4048,23 +4057,7 @@ namespace UnitarySystems {
             }
         }
 
-        if (AirLoopNumber == 0 && !ZoneEquipmentFound &&
-            (this->m_ControlType == UnitarySysCtrlType::Load || this->m_ControlType == UnitarySysCtrlType::CCMASHRAE)) {
-            ShowSevereError(state, format("{} = {}", cCurrentModuleObject, thisObjectName));
-            ShowContinueError(state, "Did not find proper connection for AirLoopHVAC or ZoneHVAC system.");
-            ShowContinueError(
-                state, format("specified Controlling Zone or Thermostat Location name = {}", input_data.controlling_zone_or_thermostat_location));
-            if (!AirNodeFound && !ZoneEquipmentFound) {
-                ShowSevereError(state, format("{} = {}", cCurrentModuleObject, thisObjectName));
-                ShowContinueError(state, "Did not find air node (zone with thermostat).");
-                // ShowContinueError(state, "specified " + cAlphaFields(iControlZoneAlphaNum) + " = " + Alphas(iControlZoneAlphaNum));
-                ShowContinueError(state,
-                                  "Both a ZoneHVAC:EquipmentConnections object and a ZoneControl:Thermostat object must be specified for this zone.");
-            }
-            errorsFound = true;
-        }
-
-        if (!ZoneEquipmentFound)
+        if (!ZoneEquipment)
```

Comment:
If the system is zone equipment, then don't add a compset, even if there were errors that might leave `ZoneEquipmentFound=false`.

Reply:
I didn't realize this nuance, that UnitarySystem zone equipment doesn't need to TestCompSet. I see that call in baseboards, fans, HXs, etc., but no zone parent objects. So what are the rules here? I thought it was a parent child relationship but obviously there is something else. There are no diffs here so this must be correct.

Reply:
The `compset` chain does not extend up to the ZoneHVAC:EquipmentList, so the components that go on that list don't need to register themselves. I suppose it could, but there is other code to warn about zone equipment that's not on a list. The node connection checking is a separate pass that uses the node connections list (set up by `GetOnlySingleNode` and the like. As for baseboards, water and steam baseboards need to chime in because they live on a plant branch. I'm fairly certain that electric baseboards do not call compsets.